be/src/vec/columns/column_vector.h

```diff
@@ -139,7 +139,11 @@ class ColumnVector final : public COWHelper<IColumn, ColumnVector<T>> {
     using Container = PaddedPODArray<value_type>;
 
     ColumnVector() = default;
-    ColumnVector(const size_t n) : data(n) {}
+
+    // for column boolean, must have initial value!
+    ColumnVector(const size_t n)
+        requires requires { !std::is_same_v<T, UInt8>; }
+            : data(n) {}
     ColumnVector(const size_t n, const value_type x) : data(n, x) {}
```

Comment:
And also should modify resize method, to require boolean type has to call resize_fill method